src/app/public/modules/tabs/tabset.component.ts

```diff
@@ -48,31 +45,74 @@ import {
 } from './tabset-adapter.service';
 
 import {
-  SkyTabsetPermalinkParams
-} from './tabset-permalink-params';
+  SkyTabsetButtonsDisplayMode
+} from './tabset-buttons-display-mode';
+
+import {
+  SkyTabsetPermalinkService
+} from './tabset-permalink.service';
 
 import {
   SkyTabsetService
 } from './tabset.service';
 
+interface TabButtonViewModel {
+  active: boolean;
+  ariaControls: string;
+  buttonHref: string;
+  buttonId: string;
+  buttonText: string;
+  buttonTextCount: string;
+  closeable: boolean;
+  disabled: boolean;
+  tabIndex: SkyTabIndex;
+}
+
 @Component({
   selector: 'sky-tabset',
   styleUrls: ['./tabset.component.scss'],
   templateUrl: './tabset.component.html',
   providers: [
     SkyTabsetAdapterService,
+    SkyTabsetPermalinkService,
     SkyTabsetService
-  ]
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SkyTabsetComponent
-  implements AfterContentInit, AfterViewInit, OnChanges, OnDestroy {
+export class SkyTabsetComponent implements AfterViewInit, OnDestroy {
 
   /**
-   * Specifies the index of the active tab.
+   * Activates a tab by its `tabIndex` property.
    * @required
```

Comment:
Is this really required? You can load the tabset without it. The tabset will automatically make the first tab active.

Reply:
True. I'll remove the `@required`.